experimental/framework/tree-react-api/src/useTree.ts

```diff
@@ -41,3 +45,40 @@ export function useTree<TRoot extends FieldSchema>(
 
 	return typedTree.context.root as unknown as SchemaAware.TypedField<TRoot>;
 }
+
+/**
+ * React Hook to trigger invalidation of the current component if anything in the document changes.
+ */
+export function useTreeContext(document: TreeContext): void {
+	// This proof-of-concept implementation allocates a state variable this is modified
+	// when the tree changes to trigger re-render.
+	const [, setInvalidations] = React.useState(0);
+
+	// Register for tree deltas when the component mounts
+	React.useEffect(() => {
+		// Returns the cleanup function to be invoked when the component unmounts.
+		return document.on("afterChange", () => {
+			setInvalidations((invalidations) => invalidations + 1);
+		});
+	}, [document]);
+}
+
+/**
+ * React Hook to trigger invalidation of the current component if anything in the provided subtree changes.
+ * This does NOT include if this subtree is moved into a different parent!
+ */
+export function useSubtree(tree: TreeNode): void {
+	// This proof-of-concept implementation allocates a state variable this is modified
+	// when the tree changes to trigger re-render.
+	const [, setInvalidations] = React.useState(0);
```

Comment:
@noencke - IIRC, you also prototyped a 'useSubtree' with your bubble popping demo?

Reply:
[Pretty much, yes](https://github.com/noencke/shared-tree-inval/blob/45027e5cba9e6b8f95ad22f3fd5758630ca4d1e0/src/useTree.ts#L46C1-L53C2) :)